client/me/notification-settings/settings-form/settings.jsx

```diff
@@ -16,80 +15,76 @@ const streams = {
 
-		return find( get( this.props.settings, 'devices' ), {
-			device_id: parseInt( this.state.selectedStream, 10 ),
+		return find( get( settings, 'devices' ), {
+			device_id: parseInt( selectedStream, 10 ),
 		} );
 	};
 
-	render() {
-		const selectedStreamSettings = this.getSelectedStreamSettings();
+	const selectedStreamSettings = getSelectedStreamSettings();
 
-		return (
-			<div className="notification-settings-form">
-				<StreamSelector
-					selectedStream={ this.state.selectedStream }
-					onChange={ ( selectedStream ) => this.setState( { selectedStream } ) }
-					settings={ selectedStreamSettings }
+	return (
+		<div className="notification-settings-form">
+			<StreamSelector
+				selectedStream={ selectedStream }
+				onChange={ setSelectedStream }
+				settings={ selectedStreamSettings }
+			/>
+			<div className="notification-settings-form__streams">
+				<Labels settingKeys={ settingKeys } />
+				<Stream
+					key={ streams.TIMELINE }
+					blogId={ blogId }
+					stream={ streams.TIMELINE }
+					settingKeys={ settingKeys }
+					settings={ get( settings, streams.TIMELINE ) }
+					onToggle={ onToggle }
 				/>
-				<div className="notification-settings-form__streams">
-					<Labels settingKeys={ this.props.settingKeys } />
-					<Stream
-						key={ streams.TIMELINE }
-						blogId={ this.props.blogId }
-						stream={ streams.TIMELINE }
-						settingKeys={ this.props.settingKeys }
-						settings={ get( this.props.settings, streams.TIMELINE ) }
-						onToggle={ this.props.onToggle }
-					/>
-					<Stream
-						key={ streams.EMAIL }
-						blogId={ this.props.blogId }
-						stream={ streams.EMAIL }
-						settingKeys={ this.props.settingKeys }
-						settings={ get( this.props.settings, streams.EMAIL ) }
-						onToggle={ this.props.onToggle }
-					/>
-					{ this.props.devices && this.props.devices.length > 0 && (
-						<Stream
-							key={ streams.DEVICES }
-							blogId={ this.props.blogId }
-							devices={ this.props.devices }
-							settingKeys={ this.props.settingKeys }
-							settings={ get( this.props.settings, streams.DEVICES ) }
-							onToggle={ this.props.onToggle }
-						/>
-					) }
+				<Stream
+					key={ streams.EMAIL }
+					blogId={ blogId }
+					stream={ streams.EMAIL }
+					settingKeys={ settingKeys }
+					settings={ get( settings, streams.EMAIL ) }
+					onToggle={ onToggle }
+				/>
+				{ devices && devices.length > 0 && (
 					<Stream
-						key={ 'selected-stream' }
-						className={ 'selected-stream' }
-						blogId={ this.props.blogId }
-						stream={ this.state.selectedStream }
-						settingKeys={ this.props.settingKeys }
-						settings={ selectedStreamSettings }
-						onToggle={ this.props.onToggle }
+						key={ streams.DEVICES }
+						blogId={ blogId }
+						devices={ devices }
+						settingKeys={ settingKeys }
+						settings={ get( settings, streams.DEVICES ) }
+						onToggle={ onToggle }
 					/>
-				</div>
+				) }
+				<Stream
+					key={ 'selected-stream' }
```

Comment:
Nit: we could just pass it as a string, no need to pass it as an expression:

```suggestion
					key="selected-stream"
```